docs/metasploit-framework.wiki/Attacking-ADCS-ESC-Vulnerabilities.md

```diff
@@ -0,0 +1,931 @@
+# Setting Up An AD CS Target
+Follow the instructions [[here|./Setting-Up-A-AD-CS-Target.md]] to set up a AD CS server
+for testing purposes.
+
+# Introduction to AD CS Vulnerabilities
+```mermaid
+flowchart TD
+	escexp[Find vulnerable certificate templates\nvia ldap_esc_vulnerable_cert_finder] --> icpr[Issue certificates via icpr_cert]
+	icpr[Issue certificates via icpr_cert] --> ESC1{{ESC1}}
+	ESC1{{ESC1}} -- Via PKINIT --> pkinit{Authenticate to Kerberos}
+	icpr[Issue certificates via icpr_cert] --> users[Request certificates on behalf of other users]
+	users[Request certificates on behalf of other users] --> ESC2{{ESC2}}
+	users[Request certificates on behalf of other users] --> ESC3{{ESC3}}
+	ESC2{{ESC2}} -- Via PKINIT --> pkinit[Authenticate to Kerberos]
+	ESC3{{ESC3}} -- Via PKINIT --> pkinit[Authenticate to Kerberos]
+```
+
+The chart above showcases how one can go about attacking three common AD CS
+vulnerabilities, taking advantage of various flaws in how certificate templates are
+configured on an Active Directory Certificate Server.
+
+The following sections will walk through each of these steps, starting with enumerating
+certificate templates that the server has to offer and identifying those that are
+vulnerable to various misconfigurations and security flaws, followed by creating new
+certificates using these certificate templates with the `icpr_cert` Metasploit module,
+and finally using these certificates to authenticate to the domain as the domain
+administrator via Kerberos.
+
+Each certificate template vulnerability that will be discussed here has a ESC code, such
+as ESC1, ESC2, or ESC3. These ESC codes are taken from the original whitepaper that
+SpecterOps published which popularized these certificate template attacks, known as
+[Certified
+Pre-Owned](https://specterops.io/wp-content/uploads/sites/3/2022/06/Certified_Pre-Owned.pdf).
+In this paper Will Schroeder and Lee Christensen described 8 different domain escalation
+attacks that they found they could conduct via misconfigured certificate templates:
+
+- ESC1 - Domain escalation via No Issuance Requirements + Enrollable Client
+  Authentication/Smart Card Logon OID templates + CT_FLAG_ENROLLEE_SUPPLIES_SUBJECT
+- ESC2 - Domain escalation via No Issuance Requirements + Enrollable Any Purpose
+  EKU or no EKU
+- ESC3 - Domain escalation via No Issuance Requirements + Certificate Request
+  Agent EKU + no enrollment agent restrictions
+- ESC4 - Domain escalation via misconfigured certificate template access control
+- ESC5 - Domain escalation via vulnerable PKI AD Object Access Control
+- ESC6 - Domain escalation via the EDITF_ATTRIBUTESUBJECTALTNAME2 setting on CAs + No
+  Manager Approval + Enrollable Client Authentication/Smart Card Logon OID templates
+- ESC7 - Vulnerable Certificate Authority Access Control
+- ESC8 - NTLM Relay to AD CS HTTP Endpoints
+
+Later another
```

Comment:
```suggestion
Later, another
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/17399/commits/239bc42f4c4228a21dd726e08fa3273545112f7f